generator/src/main/java/com/algorand/sdkutils/listeners/JavaGenerator.java

```diff
@@ -1,4 +1,933 @@
+            }
+            JavaGenerator.append(currentModelBuffer, "public class " + className + " extends PathResponse {\n\n");
+        }
+
+        pendingOpenFile = true;
+        modelPropertyAdded = false;
+    }
+
+    // Returns an iterator in sorted order of the properties (json nodes). 
+    Iterator<Entry<String, TypeDef>> getSortedProperties() {
+        Iterator<Entry<String, TypeDef>>sortedProps = properties.entrySet().iterator();
+        return sortedProps;
+    }
+
+    // Write the properties of the Model class.
+    void writeProperties(StringBuilder buffer) {
+        Iterator<Entry<String, TypeDef>> sorted = getSortedProperties();
+        while (sorted.hasNext()) {
+            Entry<String, TypeDef> prop = sorted.next();
+            TypeDef typeObj = prop.getValue();
+
+            JavaGenerator.setImportOf(typeObj, imports);
+
+            String desc = null;
+            if (typeObj.doc != null) {
+                desc = typeObj.doc;
+                desc = Tools.formatComment(desc, TAB + "", true);
+            }
+
+            // public type
+            if (desc != null) buffer.append(desc);
+            getPropertyWithJsonSetter(typeObj, buffer);
+            buffer.append("\n");
+        }
+    }
+
+    static void writeExpandedDefinitions(TypeDef typeObj, StringBuilder buffer,
+            TreeMap<String, Set<String>> imports, boolean forModel){
+        JavaGenerator.setImportOf(typeObj, imports, forModel);
+        if (typeObj.javaTypeName.equalsIgnoreCase("Address")) {
+            getAddress(typeObj, buffer, forModel);
+        }
+
+        // Enum is not defined here. It is defined separately in a 
+        // different file.
+
+        if (typeObj.rawTypeName.equals("binary") || 
+                typeObj.rawTypeName.equals("byte")) {
+            if (typeObj.javaTypeName.equals("byte[]")) {
+                getBase64Encoded(typeObj, buffer);    
+                return;
+            }
+            getBase64EncodedArray(typeObj, buffer, forModel);
+
+        }
+
+    }
+
+    // getPropertyWithJsonSetter formats the property into java declaration type with 
+    // the appropriate json annotation.
+    void getPropertyWithJsonSetter(TypeDef typeObj, StringBuilder buffer) {
+        String jprop = typeObj.propertyName;
+        String javaName = Tools.getCamelCase(jprop, false);
+        if (typeObj.isOfType("getterSetter")) {
+            writeExpandedDefinitions(typeObj, buffer, imports, true);
+            return;
+        }
+
+        if (typeObj.isOfType("enum")) {
+            this.javagen.storeEnumDefinition(typeObj);
+        }
+        
+        switch (typeObj.javaTypeName) {
+        case "java.util.DateTime":
+            throw new RuntimeException("Parsing of time is not yet implemented!");
+        case "String":
+        case "Long":
+        case "long":
+        case "boolean":
+        case "java.math.BigInteger":
```

Comment:
Don't use `javaTypeName` here, you can use the regular type instead and delete javaTypeName from the codebase.

For example check `x-algorand-type` for `unit64` instead of `BigInteger`. Also I still believe that we should pass all relevant information to the listeners - `type`, `x-algorand-type` and `format`.

Reply:
I don't think all information should be passed to the listeners. 
The minimum amount of information necessary should be passed. And I think at this point more information is getting passed than it needs to be, but fixing this can be a simple, localized change for the future.